src/metalkit.cs

```diff
@@ -219,18 +219,21 @@ interface MTKTextureLoader {
 		void FromUrl (NSUrl url, [NullAllowed] NSDictionary options, MTKTextureLoaderCallback completionHandler);
 
 		[Wrap ("FromUrl (url, options == null ? null : options.Dictionary, completionHandler)")]
```

Comment:
Yes, that's fine - nothing but the called signature is needed for the async template. So you get
```
[CompilerGenerated]
		public unsafe Task<global::Metal.IMTLTexture> FromUrlAsync (NSUrl url, MTKTextureLoaderOptions options)
		{
			var tcs = new TaskCompletionSource<global::Metal.IMTLTexture> ();
			FromUrl(url, options, (texture, error) => {
				if (error != null)
					tcs.SetException (new NSErrorException(error));
				else
					tcs.SetResult (texture);
			});
			return tcs.Task;
		}
```